src/persist/src/s3.rs

```diff
@@ -735,6 +740,92 @@ impl Blob for S3Blob {
     }
 }
 
+/// WIP temporary artifact of fast prototyping
+#[derive(Debug, Clone)]
+pub struct S3BlobMultiWriter {
+    blob: Arc<Mutex<S3Blob>>,
+}
+
+impl S3BlobMultiWriter {
+    /// Returns a new [S3BlobMultiWriter] opened for multi-writer usage.
```

Comment:
Sanity checking the intent here: this is meant to be a struct that one can instantiate multiple times on separate processes / separate machines right and not just for multiple logical writers within one process?

Reply:
yeah, the name is a temporary artifact to distinguish it from the semantics we had before. once we rip out the old interface, stuff can just be renamed back to un-suffixed thing